src/base/RegexPattern.cc

```diff
@@ -8,34 +8,34 @@
 
 #include "squid.h"
 #include "base/RegexPattern.h"
-#include "base/TextException.h"
 #include "debug/Stream.h"
 #include "sbuf/Stream.h"
 
 #include <iostream>
-#include <utility>
 
-RegexPattern::RegexPattern(const SBuf &aPattern, const int aFlags):
-    pattern(aPattern),
-    flags(aFlags)
+RegexPattern::RegexPattern(const SBuf &aPattern, const std::regex::flag_type aFlags):
+    pattern(aPattern)
 {
-    memset(&regex, 0, sizeof(regex)); // paranoid; POSIX does not require this
-    if (const auto errCode = regcomp(&regex, pattern.c_str(), flags)) {
-        char errBuf[256];
-        // for simplicity, ignore any error message truncation
-        (void)regerror(errCode, &regex, errBuf, sizeof(errBuf));
-        // POSIX examples show no regfree(&regex) after a regcomp() error;
-        // presumably, regcom() frees any allocated memory on failures
-        throw TextException(ToSBuf("POSIX regcomp(3) failure: (", errCode, ") ", errBuf,
-                                   Debug::Extra, "regular expression: ", pattern), Here());
+    try {
+        regex = std::regex(pattern.rawContent(), pattern.length(), aFlags|std::regex::nosubs|std::regex::optimize);
+    } catch (const std::regex_error &e) {
+        throw TextException(ToSBuf("bad regular expression",
+                                   Debug::Extra, "message: ", e.what(),
```

Comment:
Again, please catch and annotate _all_ exceptions (and not just std::regex_error), be more specific about the context (i.e. what operation has failed when processing a bad regular expression), and more consistent (with the existing code) when reporting the caught exception. Here is a sketch:

```suggestion
    } catch (...) {
        throw TextException(ToSBuf("failed to compile a regular expression",
                                   Debug::Extra, "problem: ", CurrentException,
```

P.S. We could also merge CurrentException printing into the first debugs() message line. Squid currently uses both approaches. I think the overall approach in the current PR code and the above suggestion is overall better despite being more verbose: The less we assume about the structure of the caught exception message, the better. This more verbose approach will also allow us to add a "more lines to follow" marker to multiline messages if we decide that we should do that.

Reply:
This try-catch is specifically to add the `pattern` value to the string in `std::regex_error` (very specifically a syntax error in the `pattern` string). All the rest of the details you are requesting are handled by other try-catch at higher levels of config parser logic.

Reply:
The downside you are failing to notice is the ongoing bloat of cache.log

Simply put: - the only errors which need the pattern value logged are these `std::regex_error` thrown by the parser when there is a problem **in the pattern**. No other errors are related in any way to the pattern value.

Reply:
> The downside you are failing to notice is the ongoing bloat of cache.log

Adding requested details does not increase "cache.log bloat" IMO:

* It is debatable that std::regex_errors "are the only errors which need the pattern value logged". It is possible that the pattern detail will be useful in some non-regex_error contexts.
* It is clear that the other details suggested in this change requests can be useful for all exception types. This change request is not (just) about the pattern!
* Most likely, these errors are fatal configuration-time errors, not frequently repeated transaction-processing errors.

It is also clear that _excluding_ the pattern detail from exceptions other than std::regex_error requires _extra_ code.

Given all these factors, adding that pattern detail to all exceptions is the best overall solution, even if one is concerned that some errors will get a pattern detail that will be useless in those specific cases.

> No other errors are related in any way to the pattern value.

We cannot reach that conclusion because we do not know what those "other errors" might be: There is no good reason for this catch statement to assume that no exception thrown by the current or future try code block can benefit from reporting the pattern (other than the std::regex_error). It is best to assume the opposite, keeping the code simple while providing _potentially_ useful context information in likely fatal configuration errors.